src/main/java/mate/academy/bookstore/handler/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,69 @@
+package mate.academy.bookstore.handler;
+
+import java.util.List;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.mapping.PropertyReferenceException;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler {
+    @ExceptionHandler(EntityNotFoundException.class)
+    @ResponseBody
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public ResponseData handleEntityNotFoundException(EntityNotFoundException ex) {
+        return new ErrorResponseData(
+                HttpStatus.NOT_FOUND,
+                null,
+                List.of(new ErrorMessage("", ex.getMessage()))
+        );
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    @ResponseBody
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ResponseData handleValidationExceptions(MethodArgumentNotValidException ex) {
+        List<ErrorMessage> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        return new ErrorResponseData(HttpStatus.BAD_REQUEST, null, errors);
+    }
+
+    @ExceptionHandler(PropertyReferenceException.class)
+    @ResponseBody
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ResponseData handlePropertyReferenceException(PropertyReferenceException ex) {
+        return new ErrorResponseData(
+                HttpStatus.BAD_REQUEST,
+                null,
+                List.of(new ErrorMessage("", ex.getMessage()))
```

Comment:
The same problem here as with ResponseData, there is not reason to use object which fields will be always null or empty. Currently ErrorMessage is working only with fields errors, I belive you should make a different object for other error cases and maybe create some hierarchy